code/game/objects/structures/curtains.dm

```diff
@@ -1,33 +1,99 @@
+/obj/item/curtain
+	name = "rolled curtain"
+	desc = "A rolled curtains."
+	icon = 'icons/obj/curtain.dmi'
+	icon_state = "curtain_rolled"
+	force = 3 //just plastic
+	w_class = ITEM_SIZE_HUGE //curtains, yeap
+	var/obj/structure/curtain/holder = /obj/structure/curtain
+
+/obj/item/curtain/attackby(obj/item/W, mob/user)
+	if(isScrewdriver(W))
+		if(!holder)
+			return
+
+		if(!isturf(loc))
+			to_chat(user, SPAN_DANGER("You cannot install \the [src] from your hands."))
+			return
+
+		if(is_space_turf(loc))
+			to_chat(user, SPAN_DANGER("You cannot install \the [src] in space."))
+			return
+
+		user.visible_message(
+			SPAN_NOTICE("[user] is installing \the [src]."),
```

Comment:
`\The [user] begins installing \the [src]/.` and `You begin installing \the [src].` would fit the various other messages a bit better.